### PR TITLE
udev: introduce Link.NamePolicy=once

### LINKED ISSUE #11436: network interface is renamed although NAME has been set by udev rule
**systemd version the issue has been seen with**

v240

**Used distribution**

Debian sid
Downstream bug report at https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=919390

To reproduce the issue, create a file `/etc/udev/rules.d/70-persistent-net.rules` containing
```
SUBSYSTEM=="net", ACTION=="add", DRIVERS=="?*", ATTR{address}=="<MAC>", KERNEL=="eth*", NAME="lan0"
```
with the mac address of your ethernet network interface.
Unload the network module (in my case `8139cp`), then load it again.
Notice how the interface is properly renamed:
```
[ 3750.870434] 8139cp 0000:00:03.0 lan0: renamed from eth0
```
Now run `udevadm trigger --action=add`
```
[ 3752.509458] 8139cp 0000:00:03.0 ens3: renamed from lan0
```
The interface is renamed although a custom NAME has been set.

### PR DESCRIPTION
**Pull request by yuwata**
An implementation of the proposal https://github.com/systemd/systemd/issues/11436#issuecomment-454585896.

Closes #11436.

(Not tested. Need to update documents.)

### CODE REVIEW

File: src/udev/net/link-config.c [hunk 1]
```diff
@@ -310,6 +310,12 @@ static bool should_rename(sd_device *device, bool respect_predictable) {
                 return true;
 
         switch (type) {
+        case NET_NAME_USER:
+        case NET_NAME_RENAMED:
+                if (once)
```
Comment (filbranden) [inline src/udev/net/link-config.c:315]: `return !once;` ?

**Comment by keszybz** [comment]
I would expect "once" or "not-once" to be completely orthogonal to the rest of the naming policy. I.e. I think it should be a flag that applies to any of the naming policies (except "kernel"). This flag says whether to apply the policy, but not what the policy actually is.

**Comment by poettering** [comment]
Yeah, I agree with @keszybz, this should be orthogonal to the naming policy, i.e. a new bool option or so. Not sure about the name though. `RenameOnce=yes|no`? or the inverse `RenameAlways=`? Or maybe `RenameOnceOnly=`? Ideas?

**Comment by yuwata** [comment]
OK. Only reason why I reuse NemePolicy= is that I could not find any good name for the new boolean :-)

**Comment by keszybz** [comment]
I started working on an update of your patch. Sorry to barge in. I'll post it soon as a separate PR.

**Pull request closed without merging**